src/modules/cmdpal/WindowsCommandPalette/Views/ListPage.xaml.cs

```diff
@@ -67,6 +68,57 @@ private bool MoreCommandsAvailable
     public ListPage()
     {
         this.InitializeComponent();
+
+        this.ItemsList.Loaded += ItemsList_Loaded;
+    }
+
+    private void ItemsList_Loaded(object sender, RoutedEventArgs e)
+    {
+        // Find the ScrollViewer in the ListView
+        var listViewScrollViewer = FindScrollViewer(this.ItemsList);
+
+        if (listViewScrollViewer != null)
+        {
+            listViewScrollViewer.ViewChanged += ListViewScrollViewer_ViewChanged;
+        }
+    }
+
+    private void ListViewScrollViewer_ViewChanged(object? sender, ScrollViewerViewChangedEventArgs e)
+    {
+        var scrollView = sender as ScrollViewer;
+        if (scrollView == null)
+        {
+            return;
+        }
+
+        // When we get to the bottom, request more from the extension, if they
+        // have more to give us.
+        // We're checking when we get to 80% of the scroll height, to give the
+        // extension a bit of a heads-up before the user actually gets there.
+        if (scrollView.VerticalOffset >= (scrollView.ScrollableHeight * .8))
+        {
+            ViewModel?.LoadMoreIfNeeded();
+        }
+    }
+
+    private ScrollViewer? FindScrollViewer(DependencyObject parent)
+    {
+        if (parent is ScrollViewer)
+        {
+            return (ScrollViewer)parent;
+        }
+
+        for (var i = 0; i < VisualTreeHelper.GetChildrenCount(parent); i++)
+        {
+            var child = VisualTreeHelper.GetChild(parent, i);
+            var result = FindScrollViewer(child);
+            if (result != null)
+            {
+                return result;
+            }
+        }
+
+        return null;
```

Comment:
FYI, we have helpers in the WCT for this, so if you're in C# never write these again 😋

```cs
ItemsList.FindAscendent<ScrollViewer>();
```